### DISCUSSION TITLE
How to force min to save a password?

### DISCUSSION THREAD

**Discussion by gundy818** [minbrowser/min#1802] [General]
When I access https://swyftx.com/au/ using firefox, it saves and auto-completes my password. When I access it in min, it doesn't, and I have to enter the password manually on every visit. How do I tell min to save the password for this site?

Also, on this site (and others) a key icon appears in both the username and password fields. Is that from the site itself, or is it something that is part of min? Clicking on the keys doesn't seem to do anything.

min is great by the way, thank you all!

**Comment by PalmerAL** [comment]
I've fixed the issue for that specific site in c774fbfb4338c58963266ff2326c8a20ef78eff4; it should show the password save prompt now. But there isn't a way to manually add passwords yet.

The key icons are from Min; if there are saved credentials for that site, clicking on one of the key icons will autofill them.

**Reply by gundy818** [reply]
Thank you for the response. I manually made that 9-line addition to the passwordFill.js in my installed version, but it doesn't seem to change anything. When I go to the site I have to type my password as usual, and when I exit and re-run min there don't seem to be any creds saved (the 'View saved passwords' link is not clickable). I noticed this is the console though:

Error occurred in handler for 'keychainFindCredentials': [Error: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.secrets was not provided by any .service files]

Is this fatal? Does it need the gnome keyring when using the Built-in password manager? I installed gnome-keyring and now I can see min querying it on dbus, so when I click 'view saved passwords' I now get a popup saying there are no passwords saved.

Thanks again.

**Reply by PalmerAL** [reply]
You do need keyring, yes. After editing the JS files, you need to recompile (since the script that actually gets run is in dist/preload.js), but I don't think the installer includes all the files needed to do so. The easiest way to test it would probably be to use the development instructions in the readme: https://github.com/minbrowser/min#developing
